sdk/sdk-go/internal/selfBackendVerifier.go

```diff
@@ -0,0 +1,554 @@
+				for i := 0; i < 4; i++ {
+					forbiddenCountriesListPacked[i] = publicSignals[discloseIndices.ForbiddenCountriesListPackedIndex+i]
+				}
+				forbiddenCountriesList = self.UnpackForbiddenCountriesList(forbiddenCountriesListPacked)
+
+				var err error
+				genericDiscloseOutput, err = self.FormatRevealedDataPacked(attestationId, publicSignals)
+				if err != nil {
+					return nil, fmt.Errorf("error formatting revealed data: %v", err)
+				}
+			}
+		}
+	}
+
+	// Determine minimum age validity from validation issues
+	isMinimumAgeValid := true
+	for _, issue := range issues {
+		if issue.Type == self.InvalidMinimumAge {
+			isMinimumAgeValid = false
+			break
+		}
+	}
+
+	isOfacValid := true
+	if configErr == nil && verificationConfig.Ofac {
+		for _, ofacCheck := range genericDiscloseOutput.Ofac {
+			if !ofacCheck {
+				isOfacValid = false
+				break
+			}
+		}
+	}
+
+	return &self.VerificationResult{
+		AttestationId: attestationId,
+		IsValidDetails: self.IsValidDetails{
+			IsValid:           isProofValid,
+			IsMinimumAgeValid: isMinimumAgeValid,
+			IsOfacValid:       isOfacValid,
+		},
+		ForbiddenCountriesList: forbiddenCountriesList,
+		DiscloseOutput:         genericDiscloseOutput,
+		UserData: self.UserData{
+			UserIdentifier:  userIdentifier,
+			UserDefinedData: userDefinedData,
+		},
+	}, nil
+}
+
+// validateWithConfig performs config-based validations (forbidden countries, minimum age, timestamp, OFAC)
+// Returns the computed values for reuse in return value construction
+func (s *SelfBackendVerifier) validateWithConfig(
+	verificationConfig self.VerificationConfig,
+	publicSignals []string,
+	discloseIndices self.DiscloseIndicesEntry,
+	attestationId self.AttestationId,
+	issues *[]self.ConfigIssue,
+) ([]string, self.GenericDiscloseOutput, error) {
+	forbiddenCountriesListPacked := make([]string, 4)
+	for i := 0; i < 4; i++ {
+		forbiddenCountriesListPacked[i] = publicSignals[discloseIndices.ForbiddenCountriesListPackedIndex+i]
+	}
+
+	forbiddenCountriesList := self.UnpackForbiddenCountriesList(forbiddenCountriesListPacked)
+
+	// Check if all config excluded countries are in the circuit's forbidden list
+	isForbiddenCountryListValid := true
+	for _, country := range verificationConfig.ExcludedCountries {
+		found := false
+		for _, circuitCountry := range forbiddenCountriesList {
+			if string(country) == circuitCountry {
+				found = true
+				break
+			}
+		}
+		if !found {
+			isForbiddenCountryListValid = false
+			break
+		}
+	}
+
+	if !isForbiddenCountryListValid {
+		*issues = append(*issues, self.ConfigIssue{
+			Type: self.InvalidForbiddenCountriesList,
+			Message: fmt.Sprintf("Forbidden countries list in config does not match with the one in the circuit\nCircuit: %s\nConfig: %v",
+				strings.Join(forbiddenCountriesList, ", "), verificationConfig.ExcludedCountries),
+		})
+	}
+	genericDiscloseOutput, err := self.FormatRevealedDataPacked(attestationId, publicSignals)
```

Comment:
you're using self.FormatRevealedDataPacked again in the verify function. Instead call it in the verify function and then pass it to this function